server/schedule/operator/operator.go

```diff
@@ -249,12 +249,17 @@ func (o *Operator) Step(i int) OpStep {
 }
 
 // getStepStartTime returns the start time of the i-th step.
-func (o *Operator) getStepStartTime(step int32) time.Time {
-	startTime := o.GetStartTime()
-	if 0 < step && int(step-1) < len(o.steps) {
-		startTime = time.Unix(0, atomic.LoadInt64(&(o.stepsTime[step-1])))
+// opStep is nil if the i-th step is not found.
+func (o *Operator) getStepStartTime() (startTime time.Time, opStep OpStep) {
+	startTime = o.GetStartTime()
+	currentStep := atomic.LoadInt32(&o.currentStep)
+	if int(currentStep) < len(o.steps) {
+		opStep = o.steps[currentStep]
+		if 0 < currentStep {
```

Comment:
check 0 < currentStep before line 257?

Reply:
the start time of the first step in operator will be the create time of the operator and the time of the second step in operator is the finished time of the first step. So the `0 < currentStep` only decide the start time. And the currentStep will be zero after the opreator initial and  be the length of the steps in operator after finishing all steps, so it may exceed the length of the steps.

Reply:
maybe `currentStep > 0`, the other way is hard to read.

Reply:
How about adding a comment?

Reply:
fix